Datum/Public/Get-FileProviderData.ps1

```diff
@@ -15,25 +15,70 @@ function Get-FileProviderData {
-                '.yml'  { ConvertFrom-Yaml (Get-Content -raw $Path) -ordered | ConvertTo-Datum -DatumHandlers $DatumHandlers }
-                
-                Default { Get-Content -Raw $Path }
+            $data = switch ($File.Extension) {
+                '.psd1' {
+                    $result = Import-PowerShellDataFile -Path $File | ConvertTo-Datum -DatumHandlers $DatumHandlers
+                    Write-Output $result -NoEnumerate
+                }
+                '.json' { 
+                    $result = ConvertFrom-Json (Get-Content -Path $Path -Raw) | ConvertTo-Datum -DatumHandlers $DatumHandlers
+                    Write-Output $result -NoEnumerate
+                }
+                '.yml'  { 
+                    $result = ConvertFrom-Yaml (Get-Content -Path $Path -Raw) -Ordered | ConvertTo-Datum -DatumHandlers $DatumHandlers
+                    Write-Output $result -NoEnumerate
+                }
+                '.csv'  {
+                    $listSeparator = $MyInvocation.MyCommand.Module.PrivateData.CsvListSeparator
+                    $innerListSeparator = $MyInvocation.MyCommand.Module.PrivateData.CsvInnerListSeperator
+
+                    if (-not $listSeparator -or -not $innerListSeparator)
+                    {
+                        Write-Error "Cannot import CSV file '$Path' as the 'ListSeparator' or 'InnerListSeparator'"
+                        return
+                    }
+                    $param = @{
+                        Path        = $Path
+                        ErrorAction = 'Stop'
+                        Delimiter   = if ($listSeparator) { $listSeparator } else { (Get-UICulture).TextInfo.ListSeparator }
+                    }
+                    if ($innerListSeparator) {
+                        $param.Add('InnerDelimiter', $innerListSeparator)
+                    }
+
+                    $result = Get-Content -Path $Path | ConvertFrom-Csv -Delimiter $listSeparator | ConvertTo-Datum -DatumHandlers $DatumHandlers
+                    for ($i = 0; $i -lt $data.Count; $i++)
+                    {
+                        for ($j = 0; $j -lt $data[$i].Keys.Count; $j++)
+                        {
+                            if ($data[$i]."$(([array]$data[$i].Keys)[$j])".Contains($innerListSeparator))
+                            {
+                                $data[$i]."$(([array]$data[$i].Keys)[$j])" = $data[$i]."$(([array]$data[$i].Keys)[$j])" -split $innerListSeparator
+                            }
+                        }                        
+                    }
+                    Write-Output $result -NoEnumerate
+                }
+                Default { Get-Content -Path $Path -Raw }
             }
+
+            if ($data -is [System.Collections.IDictionary]) {
+                $data.Add('__DatumInternal_Path', $Path)
+            }
+            $data | Add-Member -Name __DatumInternal_Path -MemberType NoteProperty -Value $Path
```

Comment:
I'm ok in principle with this, but it need to be done/extended at the [Provider] level, so we can keep track of ProviderPath (The file in the FileProvider) and Data Structure path (the suffix from the Provider path to the value).
Working on this, that will be useful to trace what change is happening where.